### PR TITLE
Feature/all cities

### PR DESCRIPTION
**Pull request by Craig180885-napier**
Use case 3.1 for review.  Wasn't so easy to figure out how to filter continent etc

### CODE REVIEW

File: src/main/java/com/napier/sem/City.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package com.napier.sem;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Class representing a country record containing the data
+ * we show when displaying a country.
```
Comment (scan-lan) [inline src/main/java/com/napier/sem/City.java]: We'll need to update this comment

File: src/main/java/com/napier/sem/City.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * This consists of:
+ * code, name, continent, region, population, capital
+ */
+public class City {
+    // Private properties
+    private String code;
```
Comment (scan-lan) [inline src/main/java/com/napier/sem/City.java]: City won't have a Code field

File: src/main/java/com/napier/sem/City.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    /**
+     * Takes the result of an executed SQL query, extracts
+     * the data we need and stores it in the class properties
+     *
+     * @param result The result of a query with a city as a result
+     * @throws SQLException
+     */
+    public City(ResultSet result) throws SQLException {
+        code = result.getString("code");
```
Comment (scan-lan) [inline src/main/java/com/napier/sem/City.java]: Same again

File: src/main/java/com/napier/sem/DAO.java [hunk 4]
```diff
@@ -6,12 +6,15 @@
 import java.sql.Statement;
 import java.util.ArrayList;
 
+// Constants that can be used as parameters when calling overloaded methods
```
Comment (scan-lan) [inline src/main/java/com/napier/sem/DAO.java]: I think this is an artefact from something that didn't go in

File: src/main/java/com/napier/sem/DAO.java [hunk 5]
```diff
@@ -49,6 +51,34 @@ public DAO(Connection connection)
...
     }
 
+    /**
+     * This takes an SQL query in the form of a string and executes it against
+     * the database.  It is only for use with statements that should return
+     * country records.  It will return the countries in a list of Country
+     * objects.
+     *
+     * @param statementString The SQL statement to be executed
+     * @return An ArrayList of country objects
```
Comment (scan-lan) [inline src/main/java/com/napier/sem/DAO.java]: We need to update this comment

**Review by scan-lan — DISMISSED** [review]
just a few small comments

**Comment by Craig180885-napier** [comment]
Cool I'll make these changes.  @LukasP23 and/or @M7DLX  can approve

File: src/main/java/com/napier/sem/City.java (hunk 2 above)
Comment (Craig180885-napier) [reply src/main/java/com/napier/sem/City.java]: Yeah that was me trying something out for 3.2 etc, I'll get rid

**Review by LukasP23 — APPROVED** [review]
Sick

**Pull request merged**